src/sonic-dhcp-server/dhcp_server/common/dhcp_db_monitor.py

```diff
@@ -343,6 +344,60 @@ def _process_check(self, key, op, entry, enabled_dhcp_interfaces):
         return False
 
 
+class DhcpServerFeatureStateChecker(ConfigDbEventChecker):
+    """
+    This event checker interested in dhcp_server feature state change in FEATURE table
+    """
+    table_name = FEATURE
+
+    def __init__(self, sel, db):
+        self.table_name = FEATURE
+        ConfigDbEventChecker.__init__(self, sel, db)
+
+    def _get_parameter(self, db_snapshot):
+        return ConfigDbEventChecker.get_parameter_by_name(db_snapshot, "dhcp_server_feature_enabled")
+
+    def _process_check(self, key, op, entry, dhcp_server_feature_enabled):
+        if key != "dhcp_server":
+            return False
+        if op == "DEL":
```

Comment:
if someone deleted dhcp server feature entry then use previous setting?

Reply:
If dhcp_server_feature_enabled (Status before delete) is enabled, means dhcp_server is enabled, delete it means status  change from enabled to disabled, this check will return True indicates enabled status change
If dhcp_server_feature_enabled is disabled, means dhcp_server is disabled, delete it means status doesn't change (disabled -> disabled), this check will return False indicates enabled status doesn't change